src/FileRouter.php

```diff
@@ -47,73 +47,108 @@ public function withNamespace(string $namespace): self
 
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
-        /**
-         * @psalm-var class-string|null $controllerClass
-         */
-        $controllerClass = $this->parseController($request);
-        if ($controllerClass === null) {
-            return $handler->handle($request);
-        }
-        /** @psalm-suppress InvalidPropertyFetch */
-        $actions = $controllerClass::$actions ?? [
-            'HEAD' => 'head',
-            'OPTIONS' => 'options',
-            'GET' => 'index',
-            'POST' => 'create',
-            'PUT' => 'update',
-            'DELETE' => 'delete',
-        ];
-        $action = $actions[$request->getMethod()] ?? null;
+        $possibleEntrypoints = $this->parseRequestPath($request);
+
+        foreach ($possibleEntrypoints as $possibleEntrypoint) {
+            /**
+             * @psalm-var class-string $controllerClass
+             * @psalm-var string|null $possibleAction
+             */
+            [$controllerClass, $possibleAction] = $possibleEntrypoint;
+            if (!class_exists($controllerClass)) {
+                continue;
+            }
 
-        if ($action === null) {
-            return $handler->handle($request);
-        }
+            /** @psalm-suppress InvalidPropertyFetch */
+            $actions = $controllerClass::$actions ?? [
+                'HEAD' => 'head',
+                'OPTIONS' => 'options',
+                'GET' => 'index',
+                'POST' => 'create',
+                'PUT' => 'update',
+                'DELETE' => 'delete',
+            ];
```

Comment:
Not need when `$possibleAction` is not null.

Reply:
https://github.com/yiisoft/file-router/pull/8/commits/abf457a5cfb1ef54f758c27a4606616a3e6c72b3#diff-0fad446e9cf870f37a58ee935b71d74a1d248c95960707b489a9e230a96d742bR63